src/components/Navigation/components/OnlineStatus/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+import { useSocket } from '../../../../context/useSocket';
+
+const StatusDropdown = ({ userId }: { userId: number | null }) => {
+  const socket = useSocket();
+  const [status, setStatus] = useState<'online' | 'offline'>('online');
+
+  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newStatus = e.target.value as 'online' | 'offline';
+    setStatus(newStatus);
+    socket.emit('set-status', { userId, status: newStatus });
+  };
+
+  if (!userId) return null;
```

Comment:
Considering that `userId` is a number, it can technically be `0`, which is a valid id, but wouldn't pass this condition. So a check like this is preferred:
```suggestion
  if (typeof userId !== 'number') return null;
```
Somewhat unrelated: it's better that ids are strings instead of numbers.